app/services/common_platform/api/representation_order_creator.rb

```diff
@@ -18,45 +18,34 @@ def call
     private
 
       def call_common_platform_endpoint
-        offences_with_case_defendant_offences.each do |offence, case_defendant_offence|
-          application_reference = maat_reference
-          offence_id = offence[:offence_id]
-          status_code = offence[:status_code]
-          status_date = offence[:status_date]
-          effective_start_date = offence[:effective_start_date]
-          effective_end_date = offence[:effective_end_date]
-
-          if case_defendant_offence.application_type.present?
+        if (court_application = CourtApplication.find_by(subject_id: defendant_id))
+          offences.each do |offence|
             CommonPlatform::Api::RecordCourtApplicationRepresentationOrder.call(
-              court_application_defendant_offence: case_defendant_offence,
-              subject_id: defendant_id,
-              offence_id:,
-              status_code:,
-              application_reference:,
-              status_date:,
-              effective_start_date:,
-              effective_end_date:,
-              defence_organisation:,
+              **offence.merge(
+                court_application_id: court_application.id,
+                subject_id: defendant_id,
+                application_reference: maat_reference,
+                defence_organisation:,
+              ),
```

Comment:
After 10 min, I realised:
`**` return the unpacked result from merge and pass it into `RecordCourtApplicationRepresentationOrder.call(`  as `keyword arguments` 

That's smart, but I try to be more in the maintainable/boring/explicit side:
Maybe something like this?

```
offence.merge(
                case_defendant_offence:,
                defendant_id:,
                defence_organisation:,
                application_reference: maat_reference,
              )

CommonPlatform::Api::RecordProsecutionCaseRepresentationOrder.call(**offence)

```